x-pack/metricbeat/module/mssql/performance/performance.go

```diff
@@ -0,0 +1,121 @@
+// interface methods except for Fetch.
+type MetricSet struct {
+	mb.BaseMetricSet
+	log     *logp.Logger
+	fetcher *mssql.Fetcher
+	db      *sql.DB
+}
+
+// New creates a new instance of the MetricSet. New is responsible for unpacking
+// any MetricSet specific configuration options if there are any.
+func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
+	cfgwarn.Beta("The mssql performance metricset is beta.")
+
+	logger := logp.NewLogger("mssql.performance").With("host", base.HostData().SanitizedURI)
+
+	db, err := mssql.NewConnection(base.HostData().URI)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not create connection to db")
+	}
+
+	return &MetricSet{
+		BaseMetricSet: base,
+		log:           logger,
+		db:            db,
+	}, nil
+}
+
+// Fetch methods implements the data gathering and data conversion to the right format
+// It returns the event which is then forward to the output. In case of an error, a
+// descriptive error must be returned.
+func (m *MetricSet) Fetch(reporter mb.ReporterV2) {
+	var err error
+	var rows *sql.Rows
+	rows, err = m.db.Query("SELECT object_name, counter_name, instance_name, cntr_value FROM sys.dm_os_performance_counters WHERE counter_name = 'SQL Compilations/sec' OR counter_name = 'SQL Re-Compilations/sec' OR counter_name = 'User Connections' OR counter_name = 'Page splits/sec' OR (counter_name = 'Lock Waits/sec' AND instance_name = '_Total') OR counter_name = 'Page splits/sec' OR (object_name = 'SQLServer:Buffer Manager' AND counter_name = 'Page life expectancy') OR counter_name = 'Batch Requests/sec' OR (counter_name = 'Buffer cache hit ratio' AND object_name = 'SQLServer:Buffer Manager') OR (counter_name = 'Lock Waits/sec' and instance_name = '_Total')")
+	if err != nil {
+		reporter.Error(errors.Wrapf(err, "error closing rows"))
+		return
+	}
+	defer func() {
+		if err := rows.Close(); err != nil {
+			m.log.Error("error closing rows: %s", err.Error())
+		}
+	}()
+
+	mapStr := common.MapStr{}
+	for rows.Next() {
+		var row performanceCounter
+		if err = rows.Scan(&row.objectName, &row.counterName, &row.instanceName, &row.counterValue); err != nil {
+			reporter.Error(errors.Wrap(err, "error scanning rows"))
+			continue
+		}
+
+		//cell values contains spaces at the beginning and at the end of the 'actual' value. They must be removed.
+		row.counterName = strings.TrimSpace(row.counterName)
+		row.instanceName = strings.TrimSpace(row.instanceName)
+		row.objectName = strings.TrimSpace(row.objectName)
+
+		if row.counterName == "Buffer cache hit ratio" {
+			mapStr[row.counterName] = fmt.Sprintf("%v", float64(*row.counterValue)/100)
+		} else {
+			mapStr[row.counterName] = fmt.Sprintf("%v", *row.counterValue)
+		}
+	}
+
+	res, err := schema.Apply(mapStr)
+	if err != nil {
+		m.log.Error(errors.Wrap(err, "error applying schema"))
+		return
+	}
+
+	if isReported := reporter.Event(mb.Event{
+		MetricSetFields: res,
+	}); !isReported {
+		m.log.Warn("event not reported")
```

Comment:
We try not to use `Warn` log messages to they are not actionable. Either Info or Error. We should have a broader discussion on what we do if we get false here and the go through all metricsets and adjust it. For now I would move this to "debug" level as it does not really provide details on why it couldn't report.

Reply:
Ok. Set to debug in the last PR